lib/redis/conn.ex

```diff
@@ -19,6 +19,13 @@ defmodule Castle.Redis.Conn do
     command(["HGET", key, field]) |> decode()
   end
 
+  def hgetall(key) do
+    command(["HGETALL", key])
+    |> Enum.chunk(2)
+    |> Enum.into(%{}, fn([k, v]) -> {k, decode(v)} end)
+    |> Map.delete("_")
```

Comment:
Castle always sets this `_` redis hash key, so it can store "empty" hashes, and distinguish that from non-existent hashes.